x/gmp/module_ibc.go

```diff
@@ -0,0 +1,205 @@
+package gmp
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	cosmossdkerrors "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	capabilitytypes "github.com/cosmos/ibc-go/modules/capability/types"
+	transfertypes "github.com/cosmos/ibc-go/v8/modules/apps/transfer/types"
+	channeltypes "github.com/cosmos/ibc-go/v8/modules/core/04-channel/types"
+	porttypes "github.com/cosmos/ibc-go/v8/modules/core/05-port/types"
+	ibcexported "github.com/cosmos/ibc-go/v8/modules/core/exported"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/sagaxyz/ssc/x/gmp/types"
+)
+
+// Message is attached in ICS20 packet memo field
+type Message struct {
+	SourceChain   string `json:"source_chain"`
+	SourceAddress string `json:"source_address"`
+	Payload       []byte `json:"payload"`
+	Type          int64  `json:"type"`
```

Comment:
Shouldn't it be MessageType?

If so, it should probably be just uint8 considering the possible values.

Reply:
Yep, good point, I followed the example integration and it was declared and not used there. The int64 type also comes from the example and I think that's how it's defined in Axelar core.